astroquery/mast/core.py

```diff
@@ -1270,6 +1278,91 @@ def _download_curl_script(self, products, out_dir):
                           "URL": [url]})
         return manifest
 
+    def enable_s3_hst_dataset(self):
+        """
+        Attempts to enable downloading HST public files from S3 instead of MAST.
+        Requires the boto3 library to function.
+        """
+        import boto3
+        self._boto3 = boto3
+
+        log.info("Using the S3 HST public dataset")
+        log.warning("Your AWS account will be charged for access to the S3 bucket")
+        log.info("See Request Pricing in https://aws.amazon.com/s3/pricing/ for details")
+        log.info("If you have not configured boto3, follow the instructions here: " +
+                 "https://boto3.readthedocs.io/en/latest/guide/configuration.html")
+
+    def disable_s3_hst_dataset(self):
+        """
+        Disables downloading HST public files from S3 instead of MAST
+        """
+        self._boto3 = None
+
+    def _download_from_s3(self, dataProduct, localPath, cache=True):
+        # The following is a mishmash of BaseQuery._download_file and s3 access through boto
+
+        bkt_name = 'stpubdata'
+
+        # This is a cheap operation and does not perform any actual work yet
+        s3 = self._boto3.resource('s3')
+        s3_client = self._boto3.client('s3')
+        bkt = s3.Bucket(bkt_name)
+
+        dataUri = dataProduct['dataURI']
+        obs_id = dataProduct['obs_id']
+
+        obs_id = obs_id.lower()
+
+        # magic associations logic per Brian - 0-9/a-e we convert to 0
+        magicValues = "123456789abcde"
+        if obs_id[-1] in magicValues:
+            new_obs_id = obs_id[:-1] + "0"
+            dataUri = dataUri.replace(obs_id, new_obs_id)
+            obs_id = new_obs_id
+            log.warning("This data product's path may not have been properly identified %s" % dataUri)
+
+        bucketPath = "hst/public/" + obs_id[:4] + dataUri.replace("mast:HST/product", "")
+
+        # CAOM LIES!
```

Comment:
You answered in my previous comment here, but could you add a note saying "CAOM is the STSCI server thingy, it presently (feb 2018) reports incorrect file sizes, so we need to do XXX about that" so that when we read the code we know why this is all here?  The "we can't use the reported file size" below is part of this.

Reply:
Sure